jerry-core/ecma/builtin-objects/ecma-builtin-string.c

```diff
@@ -109,6 +110,99 @@ ecma_builtin_string_object_from_char_code (ecma_value_t this_arg, /**< 'this' ar
   return ret_value;
 } /* ecma_builtin_string_object_from_char_code */
 
+#if ENABLED (JERRY_ES2015_BUILTIN)
+/**
+ * UTF16 Encoding method for a code point
+ *
+ * See also:
+ *          ECMA-262 v6, 10.1.1
+ *
+ * @return uint8_t, the number of returning code points
+ */
+static uint8_t
+ecma_builtin_string_utf16_encode_code_point (lit_code_point_t cp, /**< the code point we encode */
+                                             lit_code_point_t *cu_p) /**< result of the encoding */
+{
+  if (cp <= LIT_UTF16_CODE_UNIT_MAX)
+  {
+    cu_p[0] = cp;
+    return 1;
+  }
+
+  cu_p[0] = (uint32_t) (floor ((cp - (LIT_UTF16_CODE_UNIT_MAX + 1)) / 1024) + LIT_UTF16_HIGH_SURROGATE_MIN);
```

Comment:
We have the `convert_code_point_to_[low/high]_surrogate` helper functions to create surrogate pairs. These should be used instead of reimplementing the logic.